rotkehlchen/chain/evm/transactions.py

```diff
@@ -428,6 +410,48 @@ def _get_erc20_transfers_for_ranges(
                 queried_ranges=[(start_ts, end_ts)],
             )
 
+    def _query_and_save_erc20_transfers_for_range(
+            self,
+            address: ChecksumEvmAddress,
+            period: TimestampOrBlockRange,
+            location_string: str,
+            update_ranges: bool = True,
+    ) -> None:
+        """Helper function to abstract ERC20 transfer querying functionality for different range types"""  # noqa: E501
+        for erc20_tx_hashes in self.evm_inquirer.etherscan.get_token_transaction_hashes(
+            account=address,
+            from_ts=Timestamp(period.from_value),
+            to_ts=Timestamp(period.to_value),
+        ):
+            for tx_hash in erc20_tx_hashes:
+                with self.database.conn.read_ctx() as cursor:
+                    tx, _ = self.get_or_create_transaction(
+                        cursor=cursor,
+                        tx_hash=tx_hash,
+                        relevant_address=address,
+                    )
+
+                if period.range_type == 'timestamps':
+                    log.debug(f'{self.evm_inquirer.chain_name} ERC20 Transfers for {address} -> update range {period.from_value} - {tx.timestamp}')  # noqa: E501
+                    if update_ranges:
```

Comment:
this if statement I'm not sure if it is needed. I'm under the impression that you always want to update the ranges queried. Is there a reason to not do it?

Reply:
answered here https://github.com/rotki/rotki/pull/9638#discussion_r2001613487